### DISCUSSION TITLE
EMP 2.0 默认构建 切回babel 的想法

### DISCUSSION THREAD

**Discussion by ckken** [empjs/emp#281] [General]
### 目前 SWC 带来的收益在
1 首次构建速度
2 cache没触达到的缓存 

## 产生的问题
1 引用库 的 sourcemap 不生效 
2 构建好的库 再编译会报错 
3 vue 3 支持不太友好 
4 社区功能较少(相对于 esbuild,esbuild 唯一问题是 es5) 

## 补充讨论 ⬇

**Comment by ckken** [comment]
考虑同时支持 react & vue

**Reply by codcer** [reply]
经手的项目, emp-cli1.x 版本,babel 性能是相当慢, 老项目迁移成本较大, 2.x看api模式定制程度提高;     @ckken 我打算基于你2.x定制化一个;在你允许的情况进行. 当下也是主要考虑swc的情况,目前还在调研中

**Reply by codcer** [reply]
当下参看这个swc,  https://github.com/pradel/create-react-app-swc/blob/main/packages/craco-swc/README.md

**Reply by ckken** [reply]
swc目前是2.0主要构建模块 不过最大问题是 不能很好适配

**Reply by codcer** [reply]
这个你们有大概的方向不,最近一直在找swc相关的一些方案;   还是考虑vite方案 + vite 下的联邦机制;

**Reply by codcer** [reply]
swc确实, babel方案下,还是dev下,效率比较低, 另外看了下umi那边, 有一个MFSU模式. 可以去踩踩.

**Reply by ckken** [reply]
可以参考下
